doc/dev/shared_state.rst

```diff
@@ -0,0 +1,44 @@
+.. _shared_state:
+
+Shared State
+============
+
+The ``shared_state`` classes are collections to store backend-specific simulator
+state and can be found in the ``backend/{multicore, gpu}`` directories.
+Functionality manipulating such items gets delegated to the implementations of
+the ``shared_state`` interface.
+
+Ions
+----
+
+Ion state is stored as a series of arrays corresponding to NMODL variables.
+These arrays have entries per CV and mechanisms need to go through the
+``node_index_`` array to map the mechanism's internal CV index to the cell
+group's CV index.
+
+=======  ======= ===============================
+Field    NMODL   Meaning
+=======  ======= ===============================
+``iX_``  ``ica`` ion current density
+``eX_``  ``eca`` ion channel reversal potential
+``Xi_``  ``cai`` internal concentration
```

Comment:
ion internal concentration and ion external concentration

Reply:
I'll remove all `ion` here. Should be clear by context.